src/Equations/ConfinementTime.cpp

```diff
@@ -55,11 +70,179 @@ real_t ConfinementTime::EvaluateParallelConfinementTime(len_t ir) {
+	
+	return 1.0 / (WOH / P + tau_inc);
+}
+
+// Odajima-Shimomura scaling law (offset linear) P.N. Yushmanov et al. “Scalings for tokamak energy confinement”. In: Nuclear Fusion 30.10 (Oct. 1990)
+real_t ConfinementTime::OS_OL_ConfinementTime(len_t ir) const
+{	
+	real_t Zeff = ions->GetZeff(ir);
+	
+	real_t q = eqsys->GetSettings()->GetReal("timestep/safetyfactor");
+	real_t gq = pow(3 * q * (q + 5) / ((q + 2) * (q + 7)), 0.6);
+	real_t fZ = pow(Zeff, 0.4) * pow((15.0-Zeff)/20.0, 0.6);
+	real_t P = GetOhmicPower(ir) * ConversionFactor;
+	real_t WOH = fZ*gq*Goldstone_scaling(GetCoeff(), ir, 1e-20);
+	real_t tau_inc = Goldstone_scaling({0.085, nullset, nullset, nullset, 2.0, nullset, 1.0, 0.5, nullset}, ir, Conv_n20);
+	
+	return 1.0 / (WOH / P + tau_inc);
+}
+
+// Rebut-Lallia scaling law (offset linear) P.N. Yushmanov et al. “Scalings for tokamak energy confinement”. In: Nuclear Fusion 30.10 (Oct. 1990)
+real_t ConfinementTime::RL_OL_ConfinementTime(len_t ir) const
+{
+	real_t Zeff = ions->GetZeff(ir);
+	
+	real_t fZ = pow(Zeff, 0.25);
+	real_t P = GetOhmicPower(ir) * ConversionFactor;
+	real_t WOH = fZ*Goldstone_scaling(GetCoeff(), ir, 1e-20);
+	real_t tau_inc = pow(Zeff, -0.5) * Goldstone_scaling({0.014, nullset, 1.0, nullset, 1.0, 0.5, 0.5, 0.5, nullset}, ir, Conv_n20);
+	
+	return 1.0 / (WOH / P + tau_inc);
+}
+
+const real_t ConfinementTime::EvaluatePerpendicularConfinementTimeType(len_t ir) const
+{
+	switch(type) 
+	{
+		case STREAM::OptionConstants::CONF_TIME_BOHM :
+			return Bohm_ConfinementTime(ir);
+		case STREAM::OptionConstants::CONF_TIME_INTOR :
+			return INTOR_ConfinementTime(ir);
+		case STREAM::OptionConstants::CONF_TIME_RLW :
+			return RLW_ConfinementTime(ir);
+		case STREAM::OptionConstants::CONF_TIME_ITER89 :
+		case STREAM::OptionConstants::CONF_TIME_IPB98 :
+		case STREAM::OptionConstants::CONF_TIME_GOLDSTONE :
+		case STREAM::OptionConstants::CONF_TIME_KAYE_BIG :
+		case STREAM::OptionConstants::CONF_TIME_CY :
+			return 1.0 / Goldstone_scaling(GetCoeff(), ir, Conv_n20);
+		case STREAM::OptionConstants::CONF_TIME_ITER97 :
+		case STREAM::OptionConstants::CONF_TIME_EIV1 :
+		case STREAM::OptionConstants::CONF_TIME_EIV2 :
+			return 1.0 / Goldstone_scaling(GetCoeff(), ir);
+		case STREAM::OptionConstants::CONF_TIME_ITER89_OL :
+			return ITER89_OL_ConfinementTime(ir);
+		case STREAM::OptionConstants::CONF_TIME_OS_OL :
+			return OS_OL_ConfinementTime(ir);
+		case STREAM::OptionConstants::CONF_TIME_RL_OL :
+			return RL_OL_ConfinementTime(ir);
+		default :
+			throw DREAM::SettingsException("Unrecognized equation type for '%s' : %d", "tau_perp", type);
+	}
+}
+
+
+real_t ConfinementTime::EvaluatePerpendicularConfinementTime(len_t ir) 
+{
+	//When using different confinement time, it can be useful to mix them in order for the plasma for the plasma to continue the 'burn-through' or to complete the current ramp up to reach the 'flat-top'.
```

Comment:
I think this comment is rather unclear.  
What you are doing when you activate the 'mixed' energy confinement is to take the maximum between Bohm and another chosen scaling law so that in case Bohm confinement time is too low (as should be the case when the plasma current grows in a tokamak) that you can choose rather another law which has a better confinement time. This maximum works well also during the early phases where the parallel confinement time dominates (smallest) and the Bohm should provide a horribly large confinement time that makes the parallel confinement time dominate.  

The name 'mixing' is not super clear already. We are taking the maximum I think.
Please modify at least the comment and if not too much work the variable name

Reply:
Noted

Reply:
clarified in the current version. I do feel a need to test but this can be a different MR. Closig the issue now.